packages/language-server/src/plugins/typescript/SnapshotManager.ts

```diff
@@ -1,20 +1,30 @@
-import { DocumentSnapshot } from './DocumentSnapshot';
+import { DocumentSnapshot, SvelteSnapshotOptions } from './DocumentSnapshot';
 
 export class SnapshotManager {
-    private static managerContainer: Map<string, SnapshotManager> = new Map();
+    constructor(
+        private projectFiles: string[]
+    ) { }
 
-    static getFromTsConfigPath(tsconfigPath: string): SnapshotManager {
-        let manager = this.managerContainer.get(tsconfigPath);
+    private documents: Map<string, DocumentSnapshot> = new Map();
 
-        if (!manager) {
-            manager = new SnapshotManager();
-            this.managerContainer.set(tsconfigPath, manager);
+    updateByFileName(fileName: string, options: SvelteSnapshotOptions) {
+        if (!this.projectFiles.includes(fileName)) {
```

Comment:
Haven't tested it, but will this work if you add a new `ts` file later on? Or will such a file always be ignored even if it's in a valid place?

Reply:
Yeah. the new project file needed to be reparsed show.

Reply:
[What Vetur does](https://github.com/vuejs/vetur/blob/fdf7009cf39c5cb55a2f327042dd10ae751b56be/server/src/services/typescriptService/serviceHost.ts#L182) is to delete the old snapshot, so that the next time snapshots are searched for, it retrieves a new version. Maybe we could do something similar if the reparsing is hard/too performance heavy. For example we could mark the snapshot as "needs update" and the next time it's retrieved it's retrieved from file system again and the version is upped by one, or save the versions in another map and decide depending on that.

Reply:
I think it would also be ok if we bump it here. and when language-service requests a new snapshot cached it and add it to the white list. In this way, we don't have to separate the snapshot and the version.